test/com/google/cose/SignMessageTest.java

```diff
@@ -0,0 +1,83 @@
+package com.google.cose;
+
+import co.nstant.in.cbor.model.ByteString;
+import co.nstant.in.cbor.model.Map;
+import co.nstant.in.cbor.model.UnsignedInteger;
+import com.google.cose.utils.CborUtils;
+import java.util.Collections;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class SignMessageTest {
+  @Test
+  public void testDeserialize() {
+    SignMessage message = SignMessage.deserialize(TestUtilities.hexStringToByteArray(
+      "8441A0A054546869732069732074686520636F6E74656E742E818343A10126A1044231315840E2AEAFD40D69D19"
+          + "DFE6E52077C5D7FF4E408282CBEFB5D06CBF414AF2E19D982AC45AC98B8544C908B4507DE1E90B717C3D3"
+          + "4816FE926A2B98F53AFD2FA0F30A"));
+    Assert.assertEquals(TestUtilities.CONTENT, new String(message.getMessage()));
+    Assert.assertEquals("A0", TestUtilities.bytesToHexString(message.getProtectedHeaderBytes()));
+    Assert.assertEquals(0, message.getUnprotectedHeaders().getKeys().size());
+    Assert.assertEquals(1, message.getSignatures().size());
+
+    Signature s = message.getSignatures().get(0);
+    Assert.assertEquals("A10126", TestUtilities.bytesToHexString(s.getProtectedHeaderBytes()));
+    Assert.assertEquals(1, s.getUnprotectedHeaders().getKeys().size());
+    Assert.assertEquals(new ByteString(TestUtilities.hexStringToByteArray("3131")),
+        s.getUnprotectedHeaders().get(new UnsignedInteger(4)));
+    Assert.assertEquals("E2AEAFD40D69D19DFE6E52077C5D7FF4E408282CBEFB5D06CBF414AF2E19D982AC45AC98B"
+        + "8544C908B4507DE1E90B717C3D34816FE926A2B98F53AFD2FA0F30A",
+        TestUtilities.bytesToHexString(s.getSignature()));
+  }
+
+  @Test
+  public void testDeserialize2() {
```

Comment:
Please give these slightly more descriptive names. Maybe `testDeserializeWithEmptyProtectedHeaderString` and `testDeserializeWithEmptyProtectedHeaderMap` or something to indicate the difference. (yeah, those are long names, I'm open to better options).

Reply:
done